azure/Guidelines.md

```diff
@@ -747,18 +747,70 @@ The ability to retry failed requests for which a client never received a respons
 <!-- Open API Spec -->
 [OpenAPI Specification]: https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md
 
-### Long Running Operations & Jobs
+### Long-Running Operations & Jobs
 
-Azure generally follows the [Microsoft REST API guidelines for Long running operations](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#13-long-running-operations).  Follow the Microsoft guidelines with the modifications / extensions given here when designing / implementing long running operations.
+When the processing for an operation may take a significant amount of time to complete, it should be
+implemented as a _long-running operation (LRO)_. This allows clients to continue running while the
+operation is being processed. The client obtains the outcome of the operation at some later time
+through another API call.
 
-Previous Azure guidelines specified "Azure-AsyncOperation" as the name of the response header containing the operation URL,
-while the Microsoft guidelines use the name "Operation-Location".
+:white_check_mark: **DO** implement an operation as an LRO if the 99th percentile response time is greater than 1s.
 
-:white_check_mark: **DO** support both `Azure-AsyncOperation` and `Operation-Location` HEADERS, even though they are redundant so that existing SDKs and clients will continue to operate.
+There are two basic patterns that can be used for long-running operations:
+1. Resource-based long-running operations (RELO)
+2. Stepwise long-running operations
+
+:white_check_mark: **DO** use the RELO pattern when the operation is on a resource that contains a "status" property that can be used to obtain the outcome of the operation.
+
+:ballot_box_with_check: **YOU SHOULD** only use the stepwise LRO pattern when the RELO pattern is not applicable.
+
+#### Resource-based long-running operations
+
+Some common situations where the RELO pattern should be used:
+1. A "create" operation for a resource where the basic structure of the resource is created immediately and includes a status field that indicates when the create has completed, e.g. "provisioning" -> "active".
+2. An action operation for a resource where both the initiation of the action and the completion of the action cause a change to the "status" property of the resource.
+
+:white_check_mark: **DO** return a `200-OK` response, `201-Created` for create operations, from the request that initiates the operation.  The response body should contain a representation of the resource that clearly indicates that the operation has been accepted or started.
+
+:white_check_mark: **DO** support a get method on the resource that returns a representation of the resource including the status field that indicates when the operation has completed.
```

Comment:
I think RELO has this restriction -- you can't execute multiple "incompatible" operations on the resource at the same time.

Reply:
I agree that this is built into the RELO pattern but what if clients do this anyway as there is no way to prevent it?
An example is a long-running operation that subtracts money from a bank account resource and fails if the balance hits $0.00